File: Python/B/1926B-vladandshapes.py
t = int(input())

for _ in range(t):
    n = int(input())
    grid = [input() for _ in range(n)]
    min_row, max_row, min_col, max_col = n, -1, n, -1
    for i in range(n):
        for j in range(n):
            if grid[i][j] == "1":
                min_row = min(min_row, i)
                max_row = max(max_row, i)
                min_col = min(min_col, j)
                max_col = max(max_col, j)
    print("SQUARE" if max_row - min_row == max_col - min_col else "TRIANGLE")

File: C++/B/1926B-vladandshapes.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        vector<vector<char>> grid(n, vector<char>(n));
        int minRow = n, maxRow = -1, minCol = n, maxCol = -1;
        for (int i = 0; i < n; ++i)
        {
            for (int j = 0; j < n; ++j)
            {
                cin >> grid[i][j];
                if (grid[i][j] == '1')
                {
                    minRow = min(minRow, i);
                    maxRow = max(maxRow, i);
                    minCol = min(minCol, j);
                    maxCol = max(maxCol, j);
                }
            }
        }
        cout << ((maxRow - minRow == maxCol - minCol) ? "SQUARE\n" : "TRIANGLE\n");
    }
}
